### DISCUSSION TITLE
nuxt-link not working properly when using `useAsyncData`

### DISCUSSION THREAD

**Discussion by educkf** [nuxt/nuxt#18179] [Questions] [answered]
Hello everyone!

Is there any particular reason for hydration not to be working (aparently) after the deploy?

Check here: [https://semeiosis.web.app/](https://semeiosis.web.app/)

It loads pages, on network it loads other pages payload present on menu, but when navigating on the menu, it reloads the whole page as if it was plain html, but it is actually SSR deployed on firebase. It should behave as a SPA after first load, right? Fetch data and etc without a full reload, just like old nuxt was doing. I understand it should be something I am doing wrong, but as there is no error on console, I am quite lost here.

This is the project repository:

[https://github.com/educkf/semeiosis/tree/main/front](https://github.com/educkf/semeiosis/tree/main/front)

**Comment by educkf** [comment]
I was debugging and found out that, if I remove `useAsyncData` from my `app.vue` script, I get the correct behaviour. 
Something like this (on root `app.vue`  file):

```
<template>
    <div>
        <app-nav />

        <br><br><br>

        <NuxtPage/>
    </div>
</template>

<script setup>
    // empty
</script>
```

...but as soon I add it inside the script tag, the router stops working SPA-like, and works as a regular and normal <a>  tag (also on root `app.vue`  file):

```
<template>
    <div>
        <app-nav />

        <br><br><br>

        <NuxtPage/>
    </div>
</template>

<script setup>
    const { data } = await useAsyncData(
        'settings', () => $fetch('/api/settings')
    );

    console.log({ data })
</script>
```

Is there a reason for that to happen?

**Reply by educkf** [reply]
I've updated the repository code, and re-deployed the application. Now, that there is absolutely no `useAsyncData` anywhere, it works perfectly, all pages changing beautifully as they should. But if I uncomment the code on `app.js` or add it on any component or page, router stops working everywhere, and it works as it there is no vue-router.

**Comment by educkf** [comment]
I could manage to use `useAsyncData` as long it was not on `app.vue` nor any components placed directly in it.

If I call `useAsyncData` on any page inside `page` folder, SPA-like route navigation keeps working. Still needs to test it on components placed on pages.

I am not sure if it is a bug or a feature, but I still need to figure out where to place any request that always needs to happen on first request. I thought `app.vue` was the place, but looks like it is not...

**Comment by danielroe** [comment, marked as answer]
This should now be fixed with https://github.com/nuxt/framework/pull/1929. Try upgrading your Nuxt version and letting me know if not.

**Reply by educkf** [reply]
Nice, thanks a lot! Now it works perfectly :)
